### ISSUE TITLE
The code for extracting the question feature by bert

### ISSUE THREAD

**Issue by LemonQC** [doc-doc/HQGA#8]
Well, Is the model in nextqa for question features the final model? Or I still need to train? I just utilize the model to extract question feature while the results drop significantly by nearly 5 points.
![image](https://user-images.githubusercontent.com/30914380/168585931-183d4b00-fe52-4579-aa2d-23792fda5dd0.png)

**Comment by doc-doc** [comment]
For NExT-QA dataset, we did not finetune BERT ourselves but directly use the officially provided BERT feature in the NExT-QA repo. 

1. You cannot directly use this model to extract question features as it it pretrained with QA pairs. 

2. Please ensure your exracted features representations are aligned with the corresponding annotations,  eg, batch size=1..

If you do not have the aformentioned problems, you may need to pretrain it yourself to get a closer result, as I do not quit remember how this model was obtained. For pretraining, pls set the max length to 37 and pretrain for about 3 epochs, save the intermediate models and select the one that gives the best results on validation set for feature extraction.

**Comment by LemonQC** [comment]
Ok, thanks. I get one closer


| |
Shandong University
|
|
***@***.***
|
On 5/16/2022 21:19，Junbin ***@***.***> wrote：

For NExT-QA dataset, we did not finetune BERT ourselves but directly use the officially provided BERT feature in the NExT-QA repo.

You cannot directly use this model to extract question features as it it pretrained with QA pairs.

Please ensure your exracted features representations are aligned with the corresponding annotations, eg, batch size=1..

If you do not have the aformentioned problems, you may need to pretrain it yourself to get a closer result, as I do not quit remember how this model was obtained. For pretraining, pls set the max length to 37 and pretrain for about 3 epochs, save the intermediate models and select the one that gives the best results on validation set for feature extraction.
